File: ros_tutor/src/sub_string.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void chatterCallback(const std_msgs::String::ConstPtr& msg)
{
	ROS_INFO("I heard: [%s]", msg->data.c_str());
}

int main(int argc, char **argv)
{
	
	ros::init(argc, argv, "sub_string");
	
	ros::NodeHandle n;

	ros::Subscriber sub = n.subscribe("pub_string", 1000, chatterCallback);

	ros::spin();

	return 0;
}

File: ros_tutor/scripts/sub_string.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

def callback_func(msg):
	data = msg.data
	rospy.loginfo("I get data : " + data)

def main():
	rospy.loginfo("Starting String Subscriber")
	rospy.init_node("sub_string")
	rospy.Subscriber("/pub_string",String,callback_func)
	rospy.spin()

if __name__ == '__main__':
	main()